src/ipc-handlers.ts

```diff
@@ -174,124 +179,132 @@ export async function createSite(
-		path,
-		adminPassword: createPassword(),
-		port,
-		running: false,
-		phpVersion: DEFAULT_PHP_VERSION,
-		isWpAutoUpdating: wpVersion === DEFAULT_WORDPRESS_VERSION,
-		customDomain,
-		enableHttps,
-	} as const;
-
-	const server = SiteServer.create( details, { wpVersion } );
-
-	if ( isWordPressDirectory( path ) ) {
-		// If the directory contains a WordPress installation, and user wants to force SQLite
-		// integration, let's rename the wp-config.php file to allow WP Now to create a new one
-		// and initialize things properly.
-		if ( forceSetupSqlite && ( await pathExists( nodePath.join( path, 'wp-config.php' ) ) ) ) {
-			fs.renameSync(
-				nodePath.join( path, 'wp-config.php' ),
-				nodePath.join( path, 'wp-config-studio.php' )
-			);
+		if ( ( await pathExists( path ) ) && ( await isEmptyDir( path ) ) ) {
+			try {
+				await createSiteWorkingDirectory( path, wpVersion );
+			} catch ( error ) {
+				// If site creation failed, remove the generated files and re-throw the
+				// error so it can be handled by the caller.
+				await shell.trashItem( path );
+				throw error;
+			}
 		}
 
-		if ( ! ( await pathExists( nodePath.join( path, 'wp-config.php' ) ) ) ) {
-			await installSqliteIntegration( path );
-		} else {
-			await updateSiteUrl( server, getSiteUrl( details ) );
+		const port = await portFinder.getOpenPort();
+
+		const details = {
+			id: siteId || crypto.randomUUID(),
+			name: siteName || nodePath.basename( path ),
+			path,
+			adminPassword: createPassword(),
+			port,
+			running: false,
+			phpVersion: DEFAULT_PHP_VERSION,
+			isWpAutoUpdating: wpVersion === DEFAULT_WORDPRESS_VERSION,
+			customDomain,
+			enableHttps,
+		} as const;
+
+		const server = SiteServer.create( details, { wpVersion } );
+
+		if ( isWordPressDirectory( path ) ) {
+			// If the directory contains a WordPress installation, and user wants to force SQLite
+			// integration, let's rename the wp-config.php file to allow WP Now to create a new one
+			// and initialize things properly.
+			if ( forceSetupSqlite && ( await pathExists( nodePath.join( path, 'wp-config.php' ) ) ) ) {
+				fs.renameSync(
+					nodePath.join( path, 'wp-config.php' ),
+					nodePath.join( path, 'wp-config-studio.php' )
+				);
+			}
+
+			if ( ! ( await pathExists( nodePath.join( path, 'wp-config.php' ) ) ) ) {
+				await installSqliteIntegration( path );
+			} else {
+				await updateSiteUrl( server, getSiteUrl( details ) );
+			}
 		}
-	}
 
-	const parentWindow = BrowserWindow.fromWebContents( event.sender );
-	sendIpcEventToRendererWithWindow( parentWindow, 'theme-details-updating', { id: details.id } );
+		const parentWindow = BrowserWindow.fromWebContents( event.sender );
+		sendIpcEventToRendererWithWindow( parentWindow, 'theme-details-updating', { id: details.id } );
 
-	userData.sites.push( server.details );
-	sortSites( userData.sites );
-	await saveUserData( userData );
+		userData.sites.push( server.details );
```

Comment:
```suggestion
		await lockAppdata();
		const userData = await loadUserData();
		userData.sites.push( server.details );
```
I recommend acquiring the lock here. `SiteServer.create` can take a long time to resolve. Holding the lock for such a long time is risky, since it may go stale.